### PR TITLE
Fix the support of videos with different bith dept in the video reader

### PR DESCRIPTION
**Pull request by JanuszL**
- the video reader incorrectly assumes that all videos have the same bit depth.
  This PR makes sure that this is checked and that the decoder instance
  is recreated when it changes.

## Category:
**Bug fix** (*non-breaking change which fixes an issue*)

## Description:
- the video reader incorrectly assumes that all videos have the same bit depth.
  This PR makes sure that this is checked and that the decoder instance
  is recreated when it changes.

### Affected modules and functionalities:
- video reader

### Key points relevant for the review:
- NA

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
  - test_video_pipeline.py:test_10bit_vid_reconfigure
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: RVID.26

**JIRA TASK**: DALI-3625

### CODE REVIEW

**Comment by JanuszL** [comment]
https://github.com/NVIDIA/DALI_extra/pull/125

**Review by szalpal — APPROVED** [review]

File: dali/operators/reader/nvdecoder/cuvideodecoder.cc [hunk 1]
```diff
@@ -161,12 +161,16 @@ void CUVideoDecoder::reconfigure(unsigned int height, unsigned int width) {
 }
 
 int CUVideoDecoder::initialize(CUVIDEOFORMAT* format) {
-    if (decoder_) {
+    while (decoder_) {
```
Comment (klecki) [inline dali/operators/reader/nvdecoder/cuvideodecoder.cc:164]: Why `while` if there is `return` at the end of this loop?

Comment (klecki) [reply dali/operators/reader/nvdecoder/cuvideodecoder.cc:164]: Oh, I missed the break. IMO, the logic is a bit convoluted, this is a bit `goto` like :P 
Can we somehow make it into
```
if (some condition)
   FAIL
if (other condition) 
   do something;
   return 1;
if (third condition)
   clear the state & continue with config?
```

**Comment by JanuszL** [comment]
!build

**Review by klecki — APPROVED** [review]

**Pull request merged**